### PR TITLE
Serialization and deserialization of evaluators

### PR DESCRIPTION
**Pull request by pleroy**
#1922.

### CODE REVIEW

File: numerics/polynomial_evaluators_body.hpp [hunk 1]
```diff
@@ -365,13 +365,55 @@ Evaluator<Value, Argument, degree>::EvaluateDerivative(
...
+  if (evaluator == Estrin<Value, Argument, degree>::Singleton()) {
+    kind = serialization::PolynomialInMonomialBasis::Evaluator::ESTRIN;
+  } else if (evaluator == Horner<Value, Argument, degree>::Singleton()) {
+    kind = serialization::PolynomialInMonomialBasis::Evaluator::HORNER;
+  } else if (evaluator ==
+             EstrinWithoutFMA<Value, Argument, degree>::Singleton()) {
+    kind =
+        serialization::PolynomialInMonomialBasis::Evaluator::ESTRIN_WITHOUT_FMA;
+  } else {
+    /*evaluator == HornerWithoutFMA<Value, Argument, degree>::Singleton())*/
```
Comment (eggrobin) [inline numerics/polynomial_evaluators_body.hpp:383]:
```suggestion
    CHECK(evaluator == HornerWithoutFMA<Value, Argument, degree>::Singleton());
```

File: numerics/polynomial_in_monomial_basis_body.hpp [hunk 2]
```diff
@@ -516,17 +550,22 @@ ReadFromMessage(serialization::Polynomial const& message) {
...
-}
+  auto polynomial = PolynomialInMonomialBasis(coefficients, origin);
 
-template<typename Value_, typename Argument_, int degree_>
-constexpr Evaluator<Value_, Difference<Argument_>, degree_> const*
-PolynomialInMonomialBasis<Value_, Argument_, degree_>::DefaultEvaluator() {
-  if constexpr (degree_ <= 3) {
-    return Horner<Value_, Difference<Argument_>, degree_>::Singleton();
+  if (is_pre_καραθεοδωρή) {
+    CHECK_NE(evaluator, nullptr)
```
Comment (eggrobin) [inline numerics/polynomial_in_monomial_basis_body.hpp:556]: CHECK_NOTNULL?

**Review by eggrobin — APPROVED** [review]

File: numerics/polynomial_evaluators_body.hpp (hunk 1 above)
Comment (pleroy) [reply numerics/polynomial_evaluators_body.hpp:383]: As discussed, let's not do that.

File: numerics/polynomial_in_monomial_basis_body.hpp (hunk 2 above)
Comment (pleroy) [reply numerics/polynomial_in_monomial_basis_body.hpp]: Done.

**Pull request merged**